src/renderer/components/LoginPage.vue

```diff
@@ -53,97 +40,96 @@
-                                }
-                            } else {
-                                if(this.apiHostName !== defaultApiHostName) {
-                                  vi.lastError = this.$t('loginPage.errorUnknownCustomHostName')
-                                } else {
-                                  vi.lastError = this.$t('loginPage.errorUnknown')
-                                }
-                            }
-                        } else if (error.request) {
-                         vi.lastError = this.$t('loginPage.errorServerUnreachable')
-                        }
-                    });
-
-                vi.loginPending = false;
-                log.debug('login end');
-            },
-        }
+  import {remote, ipcRenderer} from "electron";
+  import {mapActions, mapState} from "vuex";
+  import EditServerAddressModal from "./LoginPage/EditServerAddressModal";
+  const log = require('electron-log');
+
+  export default {
+    name: 'login',
+    components: {EditServerAddressModal},
+    data(){
+      return {
+        windowHeight: 438,
+        email: '',
+        password: '',
+        loginPending: false,
+        lastError: '',
+        updatingServerAddress: false
+      }
+    },
+
+    mounted () {
+      const vi = this;
+      // to resize window to page content
+      const window = remote.getCurrentWindow();
+      window.setContentSize(window.getContentSize()[0], vi.windowHeight); // keep same width
+
+      // force disconnect user when app start in case disconnectUser wasn't call on close
+      this.disconnectUser(vi.$api, 'auto disconnect at startup')
+      .catch(error => log.error('disconnect failed, ignoring error:\n' + error));
+
+      // start local server to listen for Oauth2 flow redirect uri
+      ipcRenderer.on('oauthFlowSuccess', (event, code) => {
+        vi.getAndStoreAccessToken(code)
+      });
+      ipcRenderer.on('oauthFlowError', (event, error) => {
+        vi.lastError = error;
+      });
+      ipcRenderer.send('startLocalServer', this.apiHostName);
+    },
+
+    destroyed () {
+      // delete events registered in mounted to avoid multiple trigger if login page is displayed multiple time
+      ipcRenderer.removeAllListeners('oauthFlowSuccess');
+      ipcRenderer.removeAllListeners('oauthFlowError');
+    },
+
+    computed: {
+      ...mapState('auth', ['clientId', 'redirectUri', 'accessToken', 'refreshToken']),
+      ...mapState('config', ['apiHostName'])
+    },
+
+    methods: {
+      open(link) {
+          this.$electron.shell.openExternal(link)
+      },
+
+      openLoginPage(){
+        log.debug('Oauth2 flow [1]: open login page');
+
+        this.lastError = '';
+
+        this.open(
+          `${this.apiHostName}/oauth2/authorize/` +
+          `?response_type=code` +
+          `&client_id=${this.clientId}` +
+          `&redirect_uri=${this.redirectUri}` +
+          `&state=random` +
```

Comment:
89d6ee3568f79c926ce394afe09e2fbfbc03f6c9 Remove state as it would require overly complicated logic to check this state from main process (where the local server run).